bip-PSBT-SP.mediawiki

```diff
@@ -0,0 +1,187 @@
+<pre>
+  BIP: ?
+  Layer: Applications
+  Title: Sending Silent Payments with PSBTs
+  Author: Andrew Toth <andrewstoth@gmail.com>
+          Ava Chow <me@achow101.com>
+          josibake <josibake@protonmail.com>
+  Comments-Summary: No comments yet.
+  Comments-URI: TBD
+  Status: Draft
+  Type: Standards Track
+  Created: 2024-05-14
+  License: BSD-2-Clause
+</pre>
+
+==Introduction==
+
+===Abstract===
+
+This document proposes additional fields and updated role responsibilities for BIP 370 PSBTv2
+which adds support for sending to silent payments as described in BIP352.
+
+===Copyright===
+
+This BIP is licensed under the 2-clause BSD license.
+
+===Motivation===
+
+Partially Signed Bitcoin Transaction Version 2 as described in BIP 370 is not compatible with sending to silent payments as described in BIP352. In particular, the output script of a silent payment cannot be computed until after all transaction inputs have been added.
+Also, any inputs that the Signer has the private keys for must be signed with SIGHASH_ALL and all inputs must not have any scriptPubKeys with Segwit version > 1.
```

Comment:
Signing with `SIGHASH_NONE` doesn't make sense because you will have to generate the ECDH shares for all payments, but they could then be removed or new ones added. So you will have to go back the signer again. I suppose technically this could be allowed but I don't see a use for it, and it could be used by a griefer to force the party to keep going back to the signer.
Signing with `SIGHASH_SINGLE` would allow another participant to add a payment to the same scan key, but a lower lexicographic order, invalidating the `k` value in the output script.
Signing with `SIGHASH_ANYONECANPAY` is unsafe as per BIP 352.

I suppose I could add this as a footnote.
But also, I don't think this sentence should be in the motivation section.

Reply:
Removed the sentence, but also added a footnote in the Signer section to clarify why.